__tests__/app.test.js

```diff
@@ -449,4 +449,102 @@ describe("app()", () => {
+            });
+            expect(articles).toHaveLength(12);
+          });
+      });
+
+      test("GET 400: should return a 400 status topic not valid if topic not valid", () => {
+        return request(app)
+          .get("/api/articles?topic=hello")
+          .expect(400)
+          .then((data) => {
+            const { body } = data;
+            expect(body.msg).toBe("not a valid topic");
+          });
+      });
+    });
+    describe("GET /api/articles?sort_by", () => {
+      test("GET 200: should return a 200 status and articles sorted by query", () => {
+        return request(app)
+          .get("/api/articles?sort_by=title")
+          .expect(200)
+          .then((data) => {
+            const { body } = data;
+            const articles = body.articles;
+            expect(articles).toBeSortedBy("title", { descending: true });
+          });
+      });
+
+      test("GET 200: sort_by query should work with different columns", () => {
+        return request(app)
+          .get("/api/articles?sort_by=author")
+          .expect(200)
+          .then((data) => {
+            const { body } = data;
+            const articles = body.articles;
+            expect(articles).toBeSortedBy("author", { descending: true });
+          });
+      });
+
+      test("GET 400: should return a 400 status can't sort by that method if given non valid column", () => {
+        return request(app)
+          .get("/api/articles?sort_by=hello")
+          .expect(400)
+          .then((data) => {
+            const { body } = data;
+            expect(body.msg).toBe("can't sort by that method");
+          });
+      });
+    });
+    describe("GET /api/articles?order", () => {
+      test("GET 200: should return a 200 status and order articles by ascending order", () => {
+        return request(app)
+          .get("/api/articles?order=asc")
+          .expect(200)
+          .then((data) => {
+            const { body } = data;
+            const articles = body.articles;
+            expect(articles).toBeSortedBy("created_at", { ascending: true });
+          });
+      });
+
+      test("GET 200: should return a 200 status and order articles by descending order", () => {
+        return request(app)
+          .get("/api/articles?order=desc")
+          .expect(200)
+          .then((data) => {
+            const { body } = data;
+            const articles = body.articles;
+            expect(articles).toBeSortedBy("created_at", { descending: true });
+          });
+      });
+
+      test("GET 400: should return a 400 status invalid order if invalid order queried", () => {
+        return request(app)
+          .get("/api/articles?order=hello")
+          .expect(400)
+          .then((data) => {
+            const { body } = data;
+            expect(body.msg).toBe(
+              "invalid order only accepts asc(ascending) or desc (descending)"
+            );
+          });
+      });
+    });
+  });
```

Comment:
I recommend another test where you use two queries at once, for example ?sort_by=author&order=asc.